File: solutions/go/allergies/1/allergies.go
package allergies

var allergies = map[string]int{
	"eggs":         0,
	"peanuts":      1,
	"shellfish":    2,
	"strawberries": 3,
	"tomatoes":     4,
	"chocolate":    5,
	"pollen":       6,
	"cats":         7,
}

func Allergies(allergyBitSet uint) []string {
	r := []string{}
	for name, id := range allergies {
		if allergyBitSet&(1<<id) != 0 {
			r = append(r, name)
		}
	}
	return r
}

func AllergicTo(allergyBitSet uint, allergen string) bool {
	return allergyBitSet&(1<<allergies[allergen]) != 0
}

File: solutions/python/allergies/2/allergies.py
# pylint: disable=disallowed-name
# pylint: disable=missing-class-docstring
# pylint: disable=missing-function-docstring
# pylint: disable=missing-module-docstring


from enum import Flag

Allergy = Flag(
    "Allergy",
    "eggs peanuts shellfish strawberries tomatoes chocolate pollen cats",
)


class Allergies:
    # pylint: disable=too-few-public-methods

    def __init__(self, score: int):
        self.flags = {allergy for allergy in Allergy if score & allergy.value}
        self.lst = [str(allergy.name) for allergy in self.flags]

    def allergic_to(self, item: str) -> bool:
        return Allergy[item] in self.flags
